Synchronize changed password to NL, when using the member-password page
in Koha intra.

diff --git a/members/member-password.pl b/members/member-password.pl
index 82cd642..e4b4c6d 100755
--- a/members/member-password.pl
+++ b/members/member-password.pl
@@ -20,6 +20,11 @@ use Koha::Token;

 use Koha::Patron::Categories;

+use Module::Load;
+if ( C4::Context->preference('NorwegianPatronDBEnable') && C4::Context->preference('NorwegianPatronDBEnable') == 1 ) {
+    load Koha::NorwegianPatronDB, qw( NLGetSyncDataFromBorrowernumber NLEncryptPIN NLSync);
+}
+
 use Digest::MD5 qw(md5_base64);
 use Encode qw( encode );

@@ -78,6 +83,16 @@ if ( $newpassword && !scalar(@errors) ) {
     my $uid    = $input->param('newuserid') || $bor->{userid};
     my $dbh    = C4::Context->dbh;
     if ( Koha::Patrons->find( $member )->update_password($uid, $digest) ) {
+
+        # Update borrower_sync row with hashed password and syncrhronize with NL
+        if ( C4::Context->preference('NorwegianPatronDBEnable') && C4::Context->preference('NorwegianPatronDBEnable') == 1 ) {
+            my $sync = NLGetSyncDataFromBorrowernumber( $member );
+            if ( $sync ) {
+                $sync->update({ 'syncstatus' => 'edited', 'hashed_pin' => NLEncryptPIN( $newpassword ) });
+                NLSync({ 'borrowernumber' => $member });
+            }
+        }
+
         $template->param( newpassword => $newpassword );
         if ( $destination eq 'circ' ) {
             print $input->redirect("/cgi-bin/koha/circ/circulation.pl?findborrower=$cardnumber");
